lib/modules/shouldGoBack.js

```diff
@@ -1,11 +1,11 @@
 import { isEqual } from 'lodash/lang';
 
-export default function shouldGoBack(urlHistory, url, queryParams) {
+export default function shouldGoBack(urlHistory, currentIndex, url, queryParams) {
   const existsHistoryAPI = (typeof history !== 'undefined') && history.back && history.state;
-  const existsUrlHistory = urlHistory && urlHistory.length > 1;
+  const existsUrlHistory = urlHistory && urlHistory.length > currentIndex;
```

Comment:
this works, but doesn't it mean something different?
`const existsUrlHistory = urlHistory && urlHistory.length > 1;` means that the history is atleast 2 in size
`const existsUrlHistory = urlHistory && urlHistory.length > currentIndex;` means that our current index is within the range of history.